File: domain/src/main/java/home/oleg/placenearme/repositories/VenueHistoryRepository.java
package home.oleg.placenearme.repositories;

import java.util.List;

import home.oleg.placenearme.models.DetailedVenue;
import io.reactivex.Completable;
import io.reactivex.Flowable;

public interface VenueHistoryRepository {

    Flowable<List<DetailedVenue>> getHistory();

    Completable checkOutFromCurrent();

    Flowable<Boolean> isHereNow(String venueId);

    Completable checkIn(DetailedVenue detailedVenue);

    Completable checkOut(String venueId);

    Completable remove(String venueId);

}

File: core-domain/src/main/java/home/oleg/placenearme/repositories/VenueHistoryRepository.kt
package home.oleg.placenearme.repositories

import home.oleg.placenearme.models.DetailedVenue
import io.reactivex.Completable
import io.reactivex.Flowable

interface VenueHistoryRepository {

    val history: Flowable<List<DetailedVenue>>

    fun checkOutFromCurrent(): Completable

    fun isHereNow(venueId: String): Flowable<Boolean>

    fun checkIn(detailedVenue: DetailedVenue): Completable

    fun checkOut(venueId: String): Completable

    fun remove(venueId: String): Completable

}
